### PR TITLE
[FIX] Prevent PickleError (owfile.py)

### PR DESCRIPTION
**Pull request by jerneju**
##### Issue
PickleError thrown if a column does not have a name.

##### Description of changes
Column name cannot be changed to an empty string or a string with whitespaces.

##### Includes
- [X] Code changes
- [X] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by astaric** [comment]
Could you give me the steps to reproduce this issue?

**Comment by jerneju** [comment]
Put File and Scatter Plot widget and then connect them. Then open some data and delete a column name. Click Apply button and close the window and then the application. You do not need to save your work. The Pickle Error should pop up.

**Comment by jerneju** [comment]
@BlazZupan suggested a better way to solve this problem. That is preventing a user to even write a blank column name (string with zero length or only with white spaces). I am now doing some changes in file _domaineditor.py_.

**Comment by lanzagar** [comment]
As it is, this warning should be an error and not allow the user to proceed.
But I think it is better to not burden the user with this at all. If there are columns with empty names in a file, orange could auto-set the names to Feature 1, Feature 2, ...
Also to be more certain we avoid later problems, `name` should be a required argument in Variable's constructor, which should also check that it is non-empty.

**Comment by jerneju** [comment]
@lanzagar: Orange already auto-set the empty names to Feature 1, Feature 2, etc. if there are such columns in a file.

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/218075133/

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/218561990/

**Comment by jerneju** [comment]
https://sentry.io/biolab/orange3/issues/218125246/

File: Orange/widgets/data/tests/test_owfile.py [hunk 1]
```diff
@@ -127,3 +127,15 @@ def test_file_not_found(self):
         # Open a sample dataset
         self.open_dataset("iris")
         self.assertFalse(self.widget.Error.file_not_found.is_shown())
+
+    def test_check_column_noname(self):
+        '''
```
Comment (lanzagar) [inline Orange/widgets/data/tests/test_owfile.py]: Docstrings: `'''` -> `"""`

File: Orange/widgets/utils/domaineditor.py [hunk 2]
```diff
@@ -80,7 +80,8 @@ def setData(self, index, value, role):
         row_data = self.variables[row]
         if role == Qt.EditRole:
             if col == Column.name:
-                row_data[col] = value
+                if not (value.isspace() or value == ""):
```
Comment (lanzagar) [inline Orange/widgets/utils/domaineditor.py]: This results in the above `if col == Column.name` succeeding and then the inside `if` doing nothing for empty values. But the function still returns True in the end.
Instead, using `if col == Column.name and not (value.isspace() or value == "")` would result either in the value being updated or nothing happens and function returns False. Which means that after an "incorrect" edit the Apply button does not become enabled (same as cancelling the edit with Esc).

File: Orange/widgets/utils/domaineditor.py [hunk 3]
```diff
@@ -211,9 +212,9 @@ def is_missing(x):
 
         for (name, tpe, place, _, _), (orig_var, orig_plc) in \
                 zip(variables,
-                    chain([(at, Place.feature) for at in domain.attributes],
-                          [(cl, Place.class_var) for cl in domain.class_vars],
-                          [(mt, Place.meta) for mt in domain.metas])):
+                        chain([(at, Place.feature) for at in domain.attributes],
```
Comment (lanzagar) [inline Orange/widgets/utils/domaineditor.py]: Why the additional indent here?

Comment (jerneju) [inline Orange/widgets/utils/domaineditor.py]: PYLINT demands some additional spaces.

**Pull request merged**